lib/parser/models.ex

```diff
@@ -509,28 +526,36 @@ end
       end
 
       defp merge(schema, %TEnum{} = enum) do
-        %Schema{schema | enums: Map.put(schema.enums, enum.name, enum)}
+        %Schema{schema | enums: Map.put(schema.enums, enum.name, update_name(schema, enum))}
       end
 
       defp merge(schema, %Exception{} = exc) do
-        %Schema{schema | exceptions: Map.put(schema.exceptions, exc.name, exc)}
+        %Schema{schema | exceptions: Map.put(schema.exceptions, exc.name, update_name(schema, exc))}
       end
 
       defp merge(schema, %Struct{} = s) do
-        %Schema{schema | structs: Map.put(schema.structs, s.name, s)}
+        %Schema{schema | structs: Map.put(schema.structs, s.name, update_name(schema, s))}
       end
 
       defp merge(schema, %Union{} = union) do
-        %Schema{schema | unions: Map.put(schema.unions, union.name, union)}
+        %Schema{schema | unions: Map.put(schema.unions, union.name, update_name(schema, union))}
       end
 
       defp merge(schema, %Service{} = service) do
-        %Schema{schema | services: Map.put(schema.services, service.name, service)}
+        %Schema{schema | services: Map.put(schema.services, service.name, update_name(schema, service))}
       end
 
       defp merge(schema, {:typedef, actual_type, type_alias}) do
         %Schema{schema | typedefs: Map.put(schema.typedefs, atomify(type_alias), actual_type)}
       end
+
+      defp update_name(%{module: nil}, model) do
```

Comment:
Can we think of a better name for this function? Looking at how it's called above, the idea of "name" is non-obvious and ambiguous.